src/Databases/DatabaseReplicatedWorker.cpp

```diff
@@ -44,6 +46,26 @@ bool DatabaseReplicatedDDLWorker::initializeMainThread()
                 /// NOTE It will not stop cleanup thread until DDLWorker::shutdown() call (cleanup thread will just do nothing)
                 break;
             }
+
+            if (database->db_settings.max_retries_before_automatic_recovery &&
+                database->db_settings.max_retries_before_automatic_recovery <= subsequent_errors_count)
+            {
+                String current_task_name;
+                {
+                    std::unique_lock lock{mutex};
+                    current_task_name = current_task;
+                }
+                LOG_WARNING(log, "Database got stuck at processing task {}: it failed {} times in a row with the same error. "
+                                 "Will reset digest to mark our replica as lost, and trigger recovery from the most up-to-date metadata "
+                                 "from ZooKeeper. See max_retries_before_automatic_recovery setting. The error: {}",
+                            current_task, subsequent_errors_count, last_unexpected_error);
+
+                String digest_str;
+                zookeeper->tryGet(database->replica_path + "/digest", digest_str);
```

Comment:
what if the digest is already 42?
Or is it okay to say that the chances are small enough to ignore it?

Reply:
Yes, the probability is very small (2^-64), and even if it happens, it will just continue retying the faulty DDL entry, so I think it's okay